tests/torch/nas/test_ps_controller.py

```diff
@@ -63,6 +70,52 @@ def update_train_bn_adapt_section(nncf_config, bn_adapt_section_is_called):
         nncf_config["bootstrapNAS"]["training"]["batchnorm_adaptation"]["num_bn_adaptation_samples"] = 0
 
 
+def update_train_kd_loss_section(nncf_config, knowledge_distillation_loss_is_called):
+    if knowledge_distillation_loss_is_called:
+        nncf_config["bootstrapNAS"]["training"].update(
+            {"compression": [{"algorithm": "knowledge_distillation", "type": "mse"}]}
+        )
+
+
+def run_actual(training_ctrl, model, mock_dataloader):
+    optimizer = SGD(model.parameters(), lr=1e-02, weight_decay=1e-02)
+    training_ctrl.set_training_lr_scheduler_args(optimizer, len(mock_dataloader))
+    training_ctrl.scheduler.epoch_step()
+    training_ctrl.multi_elasticity_handler.activate_minimum_subnet()
+    model.train()
+    output_storage = []
+    for _, (input_, __) in enumerate(mock_dataloader):
+        input_ = input_.to(get_model_device(model))
+        output = model(input_)
+        output_storage.append(output)
+        loss = training_ctrl.loss()
+        optimizer.zero_grad()
+        loss.backward()
+        optimizer.step()
+    return output_storage
+
+
+def run_reference(training_ctrl, model, kd_model, mock_dataloader):
+    optimizer = SGD(model.parameters(), lr=1e-02, weight_decay=1e-02)
+    training_ctrl.set_training_lr_scheduler_args(optimizer, len(mock_dataloader))
+    training_ctrl.scheduler.epoch_step()
+    training_ctrl.multi_elasticity_handler.activate_minimum_subnet()
+    mse = torch.nn.MSELoss().cuda()
+    model.train()
+    kd_model.train()
+    output_storage = []
+    for _, (input_, __) in enumerate(mock_dataloader):
+        input_ = input_.to(get_model_device(model))
+        output = model(input_)
+        kd_output = kd_model(input_)
+        output_storage.append(output)
+        loss = mse(output, kd_output)
```

Comment:
`run_actual` and `run_reference` looks identical, except calculation of the loss
why not extract this to the functor:
```python
def run_train(training_ctrl, model, kd_model, mock_dataloader, calc_loss_fn):
   ...
   loss = calc_loss_fn(output)
```

```python
def calc_loss_actual(output, training_ctrl):
    return training_ctrl.loss()
```

```python
def calc_loss_reference(output, kd_model):
    kd_output = kd_model(input_)
    return mse(output, kd_output)
```

```python
run_train(...,calc_loss_fn=partial(calc_loss_actual, training_ctrl), ..)
run_train(...,calc_loss_fn=partial(calc_loss_reference, kd_model), ..)
```

Reply:
@ljaljushkin Please check the refactoring of the test. Thanks! @yzheng124